README.md

```diff
@@ -166,16 +236,156 @@ Response
+| 409       | Blocked by another session  |
+| 500       | Unknown error by receiver   |
+
+### 4.3 Cancel
 
 This route will be called when the sender wants to cancel the session.
 
-The server will use the IP address to determine if the request is legitimate.
+Use the `sessionId` from `/send-request`.
 
-`POST /api/localsend/v1/cancel`
+`POST /api/localsend/v2/cancel?sessionId=mySessionId`
 
 Response
 
 ```text
 No body
 ```
+
+## 5. Reverse file transfer (HTTP)
+
+This is an alternative method which should be used when LocalSend is not available on the receiver.
+
+The sender setups an HTTP server to send files to other members by providing a URL.
+
+The receiver then opens the browser with the given URL and downloads the file.
+
+It is important to note that the unencrypted HTTP protocol is used because browsers reject self-signed certificates.
+
+### 5.1 Browser URL
+
+The receiver can open the following URL in the browser to download the file.
+
+```text
+http://<sender-ip>:<sender-port>
+```
+
+### 5.2 Receive Request (Metadata only)
+
+Send to the sender a request to get a list of file metadata.
+
+The downloader may add `?sessionId=mySessionId`. In this case, the request should be accepted if it is the same session.
+
+This is needed if the user refreshes the browser page.
+
+`POST /api/localsend/v2/prepare-download`
+
+Request
+
+```text
+No body
+```
+
+Response
+
+```json5
+{
+  "info": {
+    "alias": "Nice Orange",
+    "version": "2.0",
+    "deviceModel": "Samsung", // nullable
+    "deviceType": "mobile", // mobile | desktop | web | headless | server, nullable
+    "fingerprint": "random string" // ignored in HTTPS mode
+  },
+  "sessionId": "mySessionId",
+  "files": {
+    "some file id": {
+      "id": "some file id",
+      "fileName": "my image.png",
+      "size": 324242, // bytes
+      "fileType": "image/jpeg",
+      "sha256": "*sha256 hash*", // nullable
+      "preview": "*preview data*" // nullable
+    },
+    "another file id": {
+      "id": "another file id",
+      "fileName": "another image.jpg",
+      "size": 1234,
+      "fileType": "image/jpeg",
+      "sha256": "*sha256 hash*",
+      "preview": "*preview data*"
+    }
+  }
+}
+```
+
+### 5.3 Receive File
+
+The file transfer.
+
+Use the `sessionId`, the `fileId` from `/receive-request`.
+
+This route can be called in parallel.
+
+`GET /api/localsend/v2/download?sessionId=mySessionId&fileId=someFileId`
+
+Request
+
+```text
+No body
+```
+
+Response
+
+```text
+Binary data
+```
+
+## 6. Additional API
+
+### 6.1 Info
+
+This was an old route previously used for discovery. This has been replaced with `/register` which is a two-way discovery.
+
+Now this route should be only used for debugging purposes.
+
+`GET /api/localsend/v2/info`
+
+Response
+
+```json5
+{
+  "alias": "Nice Orange",
+  "version": "2.0",
+  "deviceModel": "Samsung", // nullable
+  "deviceType": "mobile", // mobile | desktop | web | headless | server, nullable
```

Comment:
https://learn.microsoft.com/en-us/windows-hardware/design/form-factors/form-factors

Reply:
I don't think this is needed